File: PrepperBox.Web/src/models/consumptionLog.ts
import { ConsumptionLogRef, ProductRef } from "./types";

/**
 * Represents a consumption log entry.
 */
interface ConsumptionLog {
    id: ConsumptionLogRef;
    productId: ProductRef;
    quantity: number;
    reason?: string;

    lastModified: number;
    dateCreated: number;
}

export default ConsumptionLog;

File: PrepperBox.Db/Models/ConsumptionLog.cs
using System.ComponentModel.DataAnnotations.Schema;
using Genius.PrepperBox.Dto.References;

namespace Genius.PrepperBox.Db.Models;

public sealed record ConsumptionLog(
    ProductRef ProductId,
    decimal Quantity,
    string? Reason) : EntityBase<int, ConsumptionLogRef>
{
    // Relations:
    [ForeignKey(nameof(ProductId))]
    public Product Product { get; init; } = null!;
}
